### PR TITLE
Cross referenced and added ARG queries

### PR DESCRIPTION
**Pull request by humblejay**
# Overview/Summary

Cross references to VM and Storage ARGs,  new query added

## Related Issues/Work Items

AB#34153

## This PR fixes/adds/changes/removes

1. Cross reference SAP-1 to VM-2  ARG query
2. Cross referenced SAP-15 to ST-1 ARG query
3. Cross referenced SAP-9 to VM-8 ARG query
4. Added SAP-8 Query
5. Removed SAP-43
 

### Breaking Changes

None

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by humblejay** [comment]
> @humblejay please include a screenshot of the successful testing of the new query.

image

File: docs/content/services/specialized-workloads/sap-on-azure/code/sap-5/sap-5.kql
Comment (ejhenry) [inline docs/content/services/specialized-workloads/sap-on-azure/code/sap-5/sap-5.kql:1]: @humblejay please ensure the query output aligns with the contribution guide on column naming and order.

Comment (humblejay) [reply docs/content/services/specialized-workloads/sap-on-azure/code/sap-5/sap-5.kql:1]: This recommendation was removed in recent update.

**Review by oZakari — CHANGES_REQUESTED** [review]
Hey @humblejay, these will be going away in APRL v2, but could you please revert the re-ordering of the recommendations, other teams reference the currently reference those IDs, so we cannot switch them. 

Also, if there are existing Azure service recommendations, please reference them within the table instead of creating an SAP specific recommendation and using an existing query from an Azure resource. [AVD](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/specialized-workloads/azure-virtual-desktop/#avd-1---use-private-link-when-connecting-to-file-share-or-key-vault) and [HPC](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/specialized-workloads/azure-hpc/) can be used as a reference point for you.

**Comment by humblejay** [comment]
> Hey @humblejay, these will be going away in APRL v2, but could you please revert the re-ordering of the recommendations, other teams reference the currently reference those IDs, so we cannot switch them.
> 
> Also, if there are existing Azure service recommendations, please reference them within the table instead of creating an SAP specific recommendation and using an existing query from an Azure resource. [AVD](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/specialized-workloads/azure-virtual-desktop/#avd-1---use-private-link-when-connecting-to-file-share-or-key-vault) and [HPC](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/specialized-workloads/azure-hpc/) can be used as a reference point for you.

The numbering is reverted, some SAP recommendations have specific resources and language although they use the same queries from VM section (SAP 1,2,9).  I am confirmed with SAP team if it is okay to drop those.

**Review by oZakari — APPROVED** [review]

**Pull request merged**